File: php/features.php
<?php
declare(strict_types=1);

// Discard SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class DiscardFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new DiscardBaseFeature();
            case "test":
                return new DiscardTestFeature();
            default:
                return new DiscardBaseFeature();
        }
    }
}

File: py/features.py
# Discard SDK feature factory

from feature.base_feature import DiscardBaseFeature
from feature.test_feature import DiscardTestFeature


def _make_feature(name):
    features = {
        "base": lambda: DiscardBaseFeature(),
        "test": lambda: DiscardTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
